### ISSUE TITLE
Add ontology annotation provenance to WOBr gene list

### ISSUE THREAD

**Issue by raymond91125** [WormBase/website#6675]
Currently, our ontology browser provides a list of annotated genes for each ontology term ([example](https://www.wormbase.org/tools/ontology_browser/show_genes?focusTermName=somatic%20gonad%20precursor&focusTermId=WBbt:0007854)).

However, users are left to figure out for themselves that based on what evidence is a gene placed on the list. To make provenance verification easier, when possible, we should provide links to the annotation objects themselves, e.g. Expr_pattern, Expr_cluster, or RNAi.

**Comment by raymond91125** [comment]
To fetch the evidence information, add a field "evidence_with" to return field list (fl) for SOLR queries.

**Comment by azurebrd** [comment]
https://github.com/WormBase/website/blob/staging/lib/WormBase/API/Service/ontology_browser.pm  Is that for go_term + phenotype + else, or just one/some of them ?

How should the script know that the first example is Expr_cluster ?  
WB:Expr#### is the pattern for Expr_pattern ?
Are those the only 2 classes that can show up in "evidence_with" ?  (There's RNAi mentioned, is that the 3rd and last type or are there other types ?  If there will be RNAi, then WB:RNAi#### ?)

**Comment by azurebrd** [comment]
There's different kinds of URLs formed if the class is 'go_term', 'phenotype', or other.  Which URLs should have this extra 'evidence_with' field ?

Okay, only WB:Expr#### and WB:WBPaper#### and throw out anything else if something else shows up.

**Comment by raymond91125** [comment]
> There's different kinds of URLs formed if the class is 'go_term', 'phenotype', or other. Which URLs should have this extra 'evidence_with' field ?
All query URLs but GO and Phenotype.
> 
> Okay, only WB:Expr#### and WB:WBPaper#### and throw out anything else if something else shows up.
Yes, if there is none, no link to speak of but the gene is still kept and displayed.

**Comment by azurebrd** [comment]
It should work the way you said you wanted it to, but the links go to the URLs you describe and don't show the objects.  Also, I don't know how to check whether non-anatomy pages are generating the correct column info.

**Comment by raymond91125** [comment]
Should I use your dev instance to view?

**Comment by azurebrd** [comment]
Ok, the example page for anatomy_term works now.  If you can test the other pages affected by this, and they're fine, then cool.  If something's wrong, let me know which pages don't look right.  If it's good, let me know to make a pull request.

**Comment by raymond91125** [comment]
Yes, the links now work.
two mods:
1. also remove WB: from the gene list display.
2. i did not realize this before, can clicking on the url opens a new window?

**Comment by azurebrd** [comment]
Both changes in, let me know how it is.

**Comment by raymond91125** [comment]
yes, works right on anatomy page. other ontology term pages don't seem to be affected. i think we are ready to request pull.

**Comment by azurebrd** [comment]
ok, cool

**Comment by raymond91125** [comment]
Thanks. I'll let you close this.

**Comment by azurebrd** [comment]
Pull request made.

**Comment by raymond91125** [comment]
Chris discovered an issue that not all annotations are shown. E.g. cah-4 should have multiple associations with anatomy term neuron but only one expr_pattern is shown. It seems 'grouping' in the query string causes this problem. We should remove "&group=true&group.field=bioentity&group.ngroups=true&group.format=simple" from lines 269 and 270.

**Comment by azurebrd** [comment]
Okay, done, but now it doesn't return anything on juancarlos.wormbase with the previous example.  I don't know what link you're looking at, nor why it's not working, just made the change as you said.

**Comment by raymond91125** [comment]
OK. I think this means that the 'group' part in the query was necessary.

**Comment by azurebrd** [comment]
So, put the whole thing back, or just parts of it ?

**Comment by raymond91125** [comment]
Revert to the original query (with grouping) and add a parameter "&group.limit=1000".
Based on Solr reference: group.limit->Specifies the number of results to return for each group. The default value is 1.

**Comment by azurebrd** [comment]
okay, done

**Comment by raymond91125** [comment]
Yes, it seems all annotations are included but they are in multiple lines. Could they be shown together such that one gene one row.

**Comment by draciti** [comment]
In the WOBr gene list: 
A useful thing would be to display the anatomy term to which the annotation was assigned (in the downloadable list of genes or in the display itself or both).

For example, I had to look up genes associated to interneuron for a request from somebody in Paul’s lab. 

If you go here:
https://www.wormbase.org/tools/ontology_browser/show_genes?focusTermName=interneuron&focusTermId=WBbt:0005113

and then click on Download genes (4726) in ‘Directly annotated + inferred) you just see the list of genes but no anatomy information is displayed. It will be useful to have there also the specific anatomy terms listed. Is that feasible?

**Comment by draciti** [comment]
No, I was wondering if it was possible to add the transitive descendant terms. e.g. for cam-1 ADAL, ADAR. If it is too much of a hassle no problem

**Comment by raymond91125** [comment]
To query for all the direct annotations of all descendents is probably not efficient in WOBr. I believe wormmine or simple mine may be better suited.
